client/client_test.go

```diff
@@ -7935,6 +7936,79 @@ func testMultipleCacheExports(t *testing.T, sb integration.Sandbox) {
 	ensureFileContents(t, filepath.Join(destDir, "unique"), string(uniqueFile))
 }
 
+// https://github.com/moby/buildkit/issues/3148
+func testMountStubsTimestamp(t *testing.T, sb integration.Sandbox) {
+	c, err := New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	const sourceDateEpoch = int64(1234567890) // Fri Feb 13 11:31:30 PM UTC 2009
+	st := llb.Image("busybox:latest").Run(
+		llb.Args([]string{"/bin/touch", fmt.Sprintf("--date=@%d", sourceDateEpoch),
+			"/bin",
+			"/etc",
+			"/tmp",
+			"/tmp/foo",
+			"/tmp/foo2",
+			"/tmp/foo2/bar",
+		}),
+		llb.AddMount("/tmp/foo", llb.Scratch(), llb.Tmpfs()),
```

Comment:
Does this test still pass if you remove this mount?

Reply:
Fails 😞 

```
    client_test.go:8008:                                                                                                                                               
                Error Trace:    client_test.go:8008                                                                                                                    
                                                        run.go:86                                                                                                      
                                                        run.go:189                                                                                                                     Error:          Not equal:                                                                                                                                                             expected: 1234567890
                                actual  : 1667970326
                Test:           TestIntegration/TestMountStubsTimestamp/worker=oci
                Messages:       tmp/
```

Reply:
I assume this has been fixed?

Reply:
Not really. Hard to cover all corner cases 😞 .
I guess it is ok to just support basic cases such as /etc for now.

Reply:
I think we should debug a bit what is happening in https://github.com/moby/buildkit/pull/3149#pullrequestreview-1179875669 . It shouldn't be that hard to figure out the multi-parent case if it is not working already. If the mountpoint doesn't exist, we should check if the parent exists as well.

Reply:
The test turned out to be wrong. Directory entries must have the '/' suffix in their names.
Will update the test.